src/overlays/actors/ovl_En_Dinofos/z_en_dinofos.c

```diff
@@ -803,261 +825,266 @@ void func_8089BD28(EnDinofos* this, PlayState* play) {
     }
 }
 
-void func_8089C024(EnDinofos* this, s32 arg1) {
-    if (arg1 == 0) {
+void EnDinofos_SetupChooseJump(EnDinofos* this, s32 jumpType) {
+    if (jumpType == DINOFOS_JUMP_TYPE_FORWARD) {
         Animation_MorphToPlayOnce(&this->skelAnime, &gDinolfosJumpAnim, 2.0f);
-        func_8089AD70(this);
+        EnDinofos_EnableBumperCollision(this);
     } else {
         Animation_MorphToPlayOnce(&this->skelAnime, &gDinolfosJumpAnim, -1.0f);
-        if ((arg1 == 2) || (arg1 == 3)) {
+        if ((jumpType == DINOFOS_JUMP_TYPE_BACKWARD) || (jumpType == DINOFOS_JUMP_TYPE_SLASH)) {
             this->colliderJntSph.base.acFlags &= ~AC_ON;
-            func_8089AD70(this);
+            EnDinofos_EnableBumperCollision(this);
         }
     }
 
     this->actor.speed = 0.0f;
-    this->unk_290 = arg1;
+    this->jumpType = jumpType;
     this->actor.world.rot.y = this->actor.shape.rot.y;
-    this->actionFunc = func_8089C0DC;
+    this->actionFunc = EnDinofos_ChooseJump;
 }
 
-void func_8089C0DC(EnDinofos* this, PlayState* play) {
+void EnDinofos_ChooseJump(EnDinofos* this, PlayState* play) {
     Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x1000);
     this->actor.world.rot.y = this->actor.shape.rot.y;
     SkelAnime_Update(&this->skelAnime);
     if (Animation_OnFrame(&this->skelAnime, 1.0f)) {
-        if (this->unk_290 == 3) {
-            func_8089C244(this);
+        if (this->jumpType == DINOFOS_JUMP_TYPE_SLASH) {
+            EnDinofos_SetupJumpSlash(this);
         } else {
-            func_8089C164(this);
+            EnDinofos_SetupJumpWithoutSlashing(this);
         }
     }
 }
 
-void func_8089C164(EnDinofos* this) {
-    if (this->unk_290 == 2) {
+void EnDinofos_SetupJumpWithoutSlashing(EnDinofos* this) {
```

Comment:
```suggestion
void EnDinofos_SetupJump(EnDinofos* this) {
```
I'll admit I'm kind of on the fence here, but the `JumpSlash` is already distinguished by `Slash` so not sure we need to add `WithoutSlashing` here.

Reply:
That is because of global renaming. When I use `EnDinofos_Jump`, it clashes with `EnDinofos_JumpExtraStuff` functions.

Reply:
Yeah, but we shouldn't need to worry about global renaming anymore right, now that it has a proper name?

Reply:
True! Will update that. Thanks.